core/arch/arm/kernel/generic_core_bootcfg.c

```diff
@@ -149,6 +146,9 @@ static struct map_area bootcfg_memory_map[] = {
 	{
 	 .type = DEVICE1_TYPE,
 	 .pa = DEVICE1_BASE, .size = DEVICE1_SIZE,
+#ifdef DEVICE1_VA_BASE
```

Comment:
Why not remove the special case for 1-to-1 mapping by defining `DEVICE1_VA_BASE` as `DEVICE1_PA_BASE` if 1-to-1 mapping is desired?